gobblin-modules/google-ingestion/src/main/java/gobblin/ingestion/google/adwords/GoogleAdWordsCredential.java

```diff
@@ -0,0 +1,179 @@
+    InputStream input = GoogleAdWordsCredential.class.getResourceAsStream("/google_adwords_credential.properties");
+    Properties prop = new Properties();
+    prop.load(input);
+
+    GoogleAdWordsCredential credential =
+        new GoogleAdWordsCredential(prop.getProperty(GoogleAdWordsSource.KEY_DEVELOPER_TOKEN),
+            prop.getProperty(prop.getProperty(ConfigurationKeys.SOURCE_ENTITY)),
+            prop.getProperty(GoogleAdWordsSource.KEY_CLIENT_ID),
+            prop.getProperty(GoogleAdWordsSource.KEY_CLIENT_SECRET));
+    String refreshToken = credential.getRefreshToken();
+    System.out.println(String.format("The updated refresh token is: %s", refreshToken));
+  }
+
+  public GoogleAdWordsCredential(WorkUnitState state) {
+    _developerToken = state.getProp(GoogleAdWordsSource.KEY_DEVELOPER_TOKEN);
+    _refreshToken = state.getProp(GoogleAdWordsSource.KEY_REFRESH_TOKEN);
+
+    _appName = state.getProp(ConfigurationKeys.SOURCE_ENTITY);
+    _clientId = state.getProp(GoogleAdWordsSource.KEY_CLIENT_ID);
+    _clientSecret = state.getProp(GoogleAdWordsSource.KEY_CLIENT_SECRET);
+  }
+
+  /**
+   * This constructor is for updating the refresh token only
+   */
+  private GoogleAdWordsCredential(String developerToken, String appName, String clientId, String clientSecret) {
+    _developerToken = developerToken;
+    _refreshToken = null;
+
+    _appName = appName;
+    _clientId = clientId;
+    _clientSecret = clientSecret;
+  }
+
+  public AdWordsSession.ImmutableAdWordsSession buildRootSession()
+      throws ValidationException, GeneralSecurityException, IOException {
+    GoogleCredential credential = buildGoogleCredential();
+    credential.setRefreshToken(_refreshToken);
+    return new AdWordsSession.Builder().withDeveloperToken(_developerToken).withOAuth2Credential(credential)
+        .buildImmutable();
+  }
+
+  private String getRefreshToken()
+      throws ValidationException, IOException, GeneralSecurityException {
+    GoogleClientSecrets clientSecrets = getGoogleClientSecrets();
+    GoogleAuthorizationCodeFlow authorizationFlow =
+        new GoogleAuthorizationCodeFlow.Builder(new NetHttpTransport(), new JacksonFactory(), clientSecrets, SCOPES)
+            // Set the access type to offline so that the token can be refreshed.
+            // By default, the library will automatically refresh tokens when it can
+            // But this can be turned off by setting api.adwords.refreshOAuth2Token=false.
+            .setAccessType("offline").build();
+
+//    GoogleCredential gc = GoogleCredential.fromStream(new FileInputStream(""))
+//        .createScoped(Collections.singletonList("SCOPE"));
+//    AdWordsSession build = new AdWordsSession.Builder().fromFile().build();
+
+    String authorizeUrl = authorizationFlow.newAuthorizationUrl().setRedirectUri(REDIRECT_URI).build();
+    System.out.println(String.format("Paste this url in your browser:%n%s%n", authorizeUrl));
```

Comment:
As before, this piece of code will be run locally in an interactive mode for OAuth2.0 authorization.